src/ucp/tag/tag_match.inl

```diff
@@ -243,7 +251,13 @@ ucp_request_recv_offload_data(ucp_request_t *req, const void *data,
      * until last fragment arrives, so it is initialized to SIZE_MAX. */
     offset = SIZE_MAX - req->recv.tag.remaining;
 
-    if (ucs_unlikely(req->recv.length < length + offset)) {
+    if (ucs_unlikely(req->recv.length < (length + offset))) {
+        /* We have to release non-contig buffer only in case of 
+         * this is not the first segment and the datatype is
```

Comment:
why only if this is not the first segment?

Reply:
@alinask since we allocate `ucp_request::rev::tag::non_contig_buffer` when handling the first segment (after the check for truncation case) and release this buffer when handling the last segment. But since the memory was not allocated yet when handling the first segment and we detect truncation case, we should release nothing. When handling other segments, we have to release memory in case of truncation.